en/concept/architecture.md

```diff
@@ -1,36 +1,121 @@
 # Architectural Overview
 
-PX4 consists of two main layers: The [PX4 flight stack](../concept/flight_stack.md), an autopilot software solution and the [PX4 middleware](../concept/middleware.md), a general robotics middleware which can support any type of autonomous robot.
+PX4 consists of two main layers: the [flight stack](#flight-stack) is an autopilot software solution,
+and the [middleware](#middleware) is a general robotics layer that can support any type of autonomous robot.
 
-All [airframes](../airframes/architecture.md), and in fact all robotic systems including boats, share a single codebase. The complete system design is [reactive](http://www.reactivemanifesto.org), which means that:
+All PX4 [airframes](../airframes/README.md) share a single codebase (this includes other robotic systems like boats, rovers, submarines etc.). The complete system design is [reactive](http://www.reactivemanifesto.org), which means that:
 
-  * All functionality is divided into exchangeable components
-  * Communication is done by asynchronous message passing 
-  * The system can deal with varying workload
+- All functionality is divided into exchangeable components
+- Communication is done by asynchronous message passing 
+- The system can deal with varying workload
 
 In addition to these runtime considerations, its modularity maximizes [reusability](https://en.wikipedia.org/wiki/Reusability).
 
+
```

Comment:
Here might be a good place for the even "higher" level system diagram I was mentioning.